internal/settings.go

```diff
@@ -47,6 +47,7 @@ type DialSettings struct {
 	ImpersonationConfig           *impersonate.Config
 	EnableDirectPath              bool
 	AllowNonDefaultServiceAccount bool
+	CustomFormat                  string
```

Comment:
I don't think we want this plumbed to the DialSettings as this is a package things and not an over all client library thing.

Also, there is no mechanism to set this field. I think I have a pattern laying around somewhere for doing something like this but it involves exposing custom client options.

Reply:
Right now DialSettings seems to already have a field called CustomClaims that only package idtoken uses

https://github.com/googleapis/google-api-go-client/blob/d6ee425a65668ee28ff97c6fb70f3497865d6572/internal/settings.go#L45

And the way it is set

https://github.com/googleapis/google-api-go-client/blob/62364a2da91e0116ee4520268eb660c2574629da/idtoken/idtoken.go#L146-L154

Reply:
You are right, but I don't think it should. This works but only because these two packages are in the same repo today, but in the future they may not be. DialSetting was used here out of convenience, but it muddies the waters of what DialSetting should be used for I think. 

There are patterns we could adopt to allow any package to declare its own options without needing to touch DialOptions at all. I think we CustomClaims could be re-written in such a manner -- as well as format.

Reply:
Do you have a pattern in mind?

My first thought is have opts be of type interface{} and then type switch each opt
```go
func NewClient(ctx context.Context, audience string, opts ...interface{}) (*http.Client, error) {
	var ds internal.DialSettings
	for _, opt := range opts {
                switch x := opt.(type) {
                case ClientOption: 
                        x.Apply(&ds)
                }
	}
``` 

Are there any immediate drawbacks you see with this?